### ISSUE TITLE
correct volume specular highlight 

### ISSUE THREAD

**Issue by wilsonCernWq** [ospray/ospray#153]
Can this be fixed? Somehow my application needs to control the specular power, but I can't do it in current ospray. It should be an easy fix although.
https://github.com/ospray/ospray/blob/devel/ospray/render/scivis/volumeIntegration.ispc#L188
https://github.com/ospray/ospray/blob/devel/ospray/render/scivis/volumeIntegration.ispc#L244

**Comment by carsonbrownlee** [comment]
hi Qi,
yes this is easy to add.  The better solution we have been waiting for is to expose the same materials as geometry to volumes, but for now I can add in shading params for the hardcoded phong in the volume.  On another note, do you still need the modification to have volume macro grids be computed per model instead of per volume or is this issue no longer relevant with the new AMR structure you have?

**Comment by wilsonCernWq** [comment]
`The better solution we have been waiting for is to expose the same materials as geometry to volumes, but for now I can add in shading params for the hardcoded phong in the volume.` Thanks

`do you still need the modification to have volume macro grids be computed per model instead of per volume or is this issue no longer relevant with the new AMR structure you have?` Actually I implemented my own volume with an option to disable `volume macro grids`, and it works currently. 
The new data format in PSAPP is no longer AMR, so it is not an urgent problem currently. Disabling marco grids is just for "backward compatibility" (so that the performance won't be too bad for some cases) and it is working right now.

Thanks :D

**Comment by johguenther** [comment]
Could you also add the specular params to your volume? I hesitate a bit to “officially” add such parameters to the volume in clear view that this is a temporary hack – it's not easy to get rid of them later...

**Comment by wilsonCernWq** [comment]
Hi Johannes
I tried what you said before, but later I realized this actually requires me to copy the whole scivis renderer into my module (because renderer needs to use the values somewhere eventually). 

This is doable but hard for maintenance (unless we somehow expose more functions in scivis renderer's ispc header files. Then I can reuse most of the functions and only rewrite the `ScivisRenderer_computeInterval` function).

**Comment by carsonbrownlee** [comment]
sorry for delay, I have put in a "ns" field into the volume.  It's true that this will need to be taken out later, however we already have a specular field added to the volume that will need to similarly be taken out.  Until we put in a proper material for the volume, this will do.

**Comment by jeffamstutz** [comment]
Let's leave this open until it's in an official release.

@wilsonCernWq for now, it's on our upstream ```devel``` branch and will be included in our upcoming v1.4.0 release....though we don't yet have a good estimate on when that will be.

**Comment by wilsonCernWq** [comment]
Thanks, I will integrate it into VisIt for now. Once it is done, I will do an update.

**Comment by carsonbrownlee** [comment]
in 1.4.0 release
